lib/puppet/http/client.rb

```diff
@@ -78,6 +92,37 @@ def put(url, headers: {}, params: {}, content_type:, body:, ssl_context: nil, us
     end
   end
 
+  def post(url, headers: {}, params: {}, content_type:, body:, ssl_context: nil, user: nil, password: nil)
+    query = encode_params(params)
+    unless query.empty?
+      url = url.dup
+      url.query = query
+    end
+
+    request = Net::HTTP::Post.new(url, @default_headers.merge(headers))
+    request.body = body
+    request['Content-Length'] = body.bytesize
+    request['Content-Type'] = content_type
+
+    execute_streaming(request, ssl_context: ssl_context, user: user, password: password) do |response|
+      response.read_body
```

Comment:
could you make this streamable like the `get` case?

Reply:
Done. I force pushed the amended commit.